### PR TITLE
Improve TupleDataset documentation

### PR DESCRIPTION
**Pull request by bkvogel**
The existing documentation seemed a bit unclear. I have tried to make it easier to understand.

### CODE REVIEW

File: chainer/datasets/tuple_dataset.py [hunk 1]
```diff
@@ -3,15 +3,27 @@
...
+
+    Recall that in Chainer, a dataset is defined as an iterable that supports
+    both ``__getitem__`` and ``__len__``. The ``__getitem__`` method should
+    support indexing by both an integer and a slice.
+
+    As an example, consider creating a ``TupleDataset`` from two argument
+    datasets ``d1 = [8, 0, 5, 1]`` and ``d2 = [3, 1, 7, 4]`` as
+    ``tuple_dataset = TupleDataset(d1, d2)``. The ``tuple_dataset`` will
+    then contain the examples ``(8,3), (0, 1), (5,7), (1,4)``. Note that this
+    behavior is similar to that of the built-in ``zip()`` function.
```
Comment (niboshi) [inline chainer/datasets/tuple_dataset.py]: ``` ``zip()`` ``` -> ``` :func:`zip` ```

File: chainer/datasets/tuple_dataset.py [hunk 2]
```diff
@@ -3,15 +3,27 @@
...
 
     Args:
-        datasets: Underlying datasets. The ``i``-th one is used for the
-            ``i``-th item of each example. All datasets must have the same
-            length.
+        datasets: Underlying datasets that will be aggregated. Each dataset
+        must be an iterable that implements ``__getitem__`` and ``__len__``.
```
Comment (niboshi) [inline chainer/datasets/tuple_dataset.py]: Put 4-space indentation before each of the following lines.

File: chainer/datasets/tuple_dataset.py [hunk 3]
```diff
@@ -3,15 +3,27 @@
...
+    each of the argument datasets, in the same order that they were supplied.
+
+    Recall that in Chainer, a dataset is defined as an iterable that supports
+    both ``__getitem__`` and ``__len__``. The ``__getitem__`` method should
+    support indexing by both an integer and a slice.
+
+    As an example, consider creating a ``TupleDataset`` from two argument
+    datasets ``d1 = [8, 0, 5, 1]`` and ``d2 = [3, 1, 7, 4]`` as
+    ``tuple_dataset = TupleDataset(d1, d2)``. The ``tuple_dataset`` will
+    then contain the examples ``(8,3), (0, 1), (5,7), (1,4)``. Note that this
```
Comment (niboshi) [inline chainer/datasets/tuple_dataset.py]: Put spaces after each comma.

Comment (bkvogel) [reply chainer/datasets/tuple_dataset.py]: Thank you for reviewing. I have made fixes based on your comments.

Comment (niboshi) [reply chainer/datasets/tuple_dataset.py]: Thank you. Please fix this one (a space after each comma), too.

**Comment by bkvogel** [comment]
I fixed  the comma issue.

**Comment by niboshi** [comment]
Thank you!

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
Jenkins test has passed. Awaiting AppVeyor

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**